### DISCUSSION TITLE
Support for latest .NET runtimes (net6, net8)

### DISCUSSION THREAD

**Discussion by Twinki14** [citizenfx/fivem#2318] [Engineering Discussion]
https://github.com/citizenfx/fivem/issues/1287 Has brought this up before and noted a potential memory corruption issue. Since that issue we've had .NET 7 and .NET 8 release, [both brought crazy-performant changes ](https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/) that FiveM might heavily benefit from, even on the serialization/deserialization area

Steve Sanderson, a dotnet developer put together this experimental library https://github.com/SteveSandersonMS/DotNetIsolator which allows you to sandbox .NET code in your .NET code via WASM

https://github.com/engineer-man/piston or going the WebAssembly route with https://github.com/extism/extism might be another sandboxing option

**Comment by nilslice** [comment]
@Twinki14 - let us know if we can help if you need it re: Extism integration. Always available at https://extism.org/discord!

**Comment by jacobwi** [comment]
Anyone working on this? I got ~ 90% migrated over to .NET 8.

**Reply by FabianTerhorst** [reply]
No one is working on it atm. But the new .NET runtime is within our scope and would be for sure a great scripting runtime. One possible way would be implementing it as a extra server only runtime for now, because of the amount of work it would require to get it sandboxed on the client.

**Reply by dotmbf** [reply]
@FabianTerhorst net8.0 for server would be a great improvement - DI in netstandard2.0 doesnt compare with standard libs from MS tbh :(

**Reply by 5pHiNxX** [reply]
Please, please, please give us net8.0 for server combined with js/ts for client

**Reply by Felipellira** [reply]
> @FabianTerhorst net8.0 for server would be a great improvement - DI in netstandard2.0 doesnt compare with standard libs from MS tbh :(
It should be implemented in both client and server, is so bad that i cannot use DI or other newer libs in the client.

**Comment by HyperExtendedReality** [comment]
New Wine flavored Mono just released(6.14.0). It's been 5 years since last release and since it has sandboxing with flatpack, sandwine, etc. Maybe FiveM can benefit from the update?
